operator/pkg/hooks/upgrade_hooks.go

```diff
@@ -161,6 +166,41 @@ func checkInitCrdJobs(kubeClient manifest.ExecClient, currentIOPS, _ *v1alpha1.I
 	return nil
 }
 
+func checkRBACCrds(kubeClient manifest.ExecClient, _, _ *v1alpha1.IstioOperatorSpec) util.Errors {
+	crds, err := kubeClient.GetCRDList()
+	if err != nil {
+		return util.NewErrs(fmt.Errorf("failed to get custom resource definitions: %v", err))
+	}
+
+	isRBACCrd := func(name string) bool {
+		rbacCrds := []string{
+			"clusterrbacconfigs.rbac.istio.io",
+			"rbacconfigs.rbac.istio.io",
+			"servicerolebindings.rbac.istio.io",
+			"serviceroles.rbac.istio.io",
+		}
+		for _, rbac := range rbacCrds {
+			if name == rbac {
+				return true
+			}
+		}
+		return false
+	}
+	var foundNames []string
+	for _, crd := range crds.Items {
+		if name := crd.Name; isRBACCrd(name) {
+			foundNames = append(foundNames, name)
+		}
+	}
+	if len(foundNames) != 0 {
+		return util.NewErrs(fmt.Errorf("found custom resource definitions (CRDs) of unsupported v1alpha1 RBAC policy: %v. "+
```

Comment:
This will be true for almost every cluster that has installed 1.4 at any point. Can we check if they actually have configs in the cluster maybe?

Reply:
I think it's better to advise users to just remove the old CRDs otherwise it will be there all the time, and this is dangerous because as long as the CRD is still there, the user  could apply the unsupported v1alpha1 RBAC policy that will just be ignored silently by us.

Also it should be easy for users to delete the CRDs if they are not using any of them with simple one line command: `kubectl delete crd clusterrbacconfigs.rbac.istio.io rbacconfigs.rbac.istio.io servicerolebindings.rbac.istio.io serviceroles.rbac.istio.io`

Reply:
I think it's better to advise users to just remove the old CRDs otherwise it will be there all the time, and this is dangerous because as long as the CRD is still there, the user  could apply the unsupported v1alpha1 RBAC policy that will just be ignored silently by us.

Also it should be easy for users to delete the CRDs if they are not using any of them with simple one line command: `kubectl delete crd clusterrbacconfigs.rbac.istio.io rbacconfigs.rbac.istio.io servicerolebindings.rbac.istio.io serviceroles.rbac.istio.io`

Reply:
do we really want to remove the alpha RBAC CRD?  If we do remove the CRD, i think the associated custom resources will be removed also, which can make users mad.

I think we should allow upgrade to proceed with warning that these won't work and should be migrated to beta resources ASAP.  I would expect us to provide warnings with istioctl analyzer cmd as well as whoever runs the upgrade cmd is likely different set of roles/users than who actually migrate the yaml to beta resources.  I'm thinking one is mesh admin or a vendor, the other is the security operator.

thoughts?

Reply:
appreciate very much you are looking at resolving this for our users @yangminzhu !

Reply:
Sure we could add the warning to `istioctl analyze` (in a separate PR), but for `upgrade` I think it really should be a hard fail, It doesn't have any benefits to keep the v1alpha1 RBAC policy in the system when upgrading to 1.6, these alpha policies will not work at all, it's probably fine if these policies are not supposed to work in the first then it's better to delete them. If these policies are supposed to protect something, we shouldn't allow users to upgrade because it will cause security holes in the mesh.

Reply:
@linsun @liminw could you take a look at this PR? After this one, I also plan to remove the alpha CRD completely and update `istioctl analyze` to give a warning on the alpha RBAC policise, thanks.